### PR TITLE
Rpc ppx syntax extension, wikidoc

### PR DESCRIPTION
**Pull request by GuerfelMohamed**

### CODE REVIEW

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 1]
```diff
@@ -0,0 +1,67 @@
+=RPC ppx syntax extension
+
+The ocsigen-start syntax extension allows to write, in one function, both the
+client and server part of your RPC.
+
+
+==Sections
+
+The location of code execution is specified by section annotations. 3 types
+of constructions are available to perform RPC sections:
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:10]: %rpc are not sections.
I would write "to build an rpc"

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 2]
```diff
@@ -0,0 +1,67 @@
+=RPC ppx syntax extension
+
+The ocsigen-start syntax extension allows to write, in one function, both the
+client and server part of your RPC.
+
+
+==Sections
+
+The location of code execution is specified by section annotations. 3 types
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:9]: "The location of code execution is specified by section annotations."

This is not the case for rpcs. You cannot choose the location of your code.

Comment (remyzorg) [reply doc/dev/api/server/Ppx_syntax.wiki:9]: Also, 
3 => Three

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+of constructions are available to perform RPC sections:
+
+<<code language="ocaml"|
+[%%rpc
+  let r = ...
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:14]: this is not a valid rpc syntax. I see what you want to do here, but you shouldn't write syntactically invalid code anyway.

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+Constructions have a shortcut:
+
+<<code language="ocaml"|
+let%rpc r = ...
+
+let%crpc c = ...
+>>
+
+
+Sections are only available at top level and cannot be nested.
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:43]: Have you tested this ? What is the error ?

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+In {{{.eliomi}}} files, an RPC is declared as shared function.
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:45]: "The corresponding declaration in {{{.eliomi}}} is located in shared code as both client code and serveur code is generated"

(or something like that)

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+==Restrictions
+
+As the RPC will be computed on both client-side and server-side,
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:49]: This is not exactly the reason.
The reason is that you have to explicite a monomorphic type to be able to translate it to json, and to communicate between server and client (on this part you are right).

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+<<code language="ocaml"|
+let%rpc r (... : <type>) = ...
+
+>>
+
+In the case of {{crpc}} and {{crpc_opt}}, as you use the {{connected_rpc}} API,
+your first RPC function argument will refer automatically to the current session 
+user id. {{crpc_opt}} annotation refers to the case when session current id 
+needed as an optional arguement:
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:61]: good

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+<<code language="ocaml"|
+let%crpc c userid (... : <type>) = ...
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:64]: use a longer name for the function. It is hard to understand.

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+let%crpc_opt userid_o (... : <type>) = ...
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:66]: function name is missing.

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+The ocsigen-start syntax extension allows to write, in one function, both the
+client and server part of your RPC.
+
+
+==Constructions
+
+Three types of constructions are available to build an RPC:
+
+<<code language="ocaml"|
+  let%rpc r = ...
```
Comment (remyzorg) [inline doc/dev/api/server/Ppx_syntax.wiki:12]: these are still not valid rpc syntax.

an rpc must have at least a parameter. Am I wrong ?

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+<<code language="ocaml"|
+  let%crpc toto userid (... = <type>) = ...
+
+  let%crpc_opt toto userid_o (... = <type>) = ...
+>>
+
+Construction to compute an RPC using the {{connected_rpc}} API.
+
+
+RPC constructions  are only available at top level. The corresponding
```
Comment (facelesspanda) [inline doc/dev/api/server/Ppx_syntax.wiki:26]:
```suggestion
RPC constructions are only available at top level. The corresponding 
```

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+
+RPC constructions  are only available at top level. The corresponding 
+declaration in {{{.eliomi}}} is located in shared section.
+
+==Restrictions
+
+Your RPC will implicitly compute a client part and server part. Explicitly 
+declare each argument monomorphic type of your RPC function will be 
+required to able json translation and communication between the two parts:
```
Comment (facelesspanda) [inline doc/dev/api/server/Ppx_syntax.wiki:33]:
```suggestion
Your RPC will implicitly compute a client part and server part. Explicit 
declaration of each argument's monomorphic type will be 
required to allow the json translation and communication between the two parts:
```

File: doc/dev/api/server/Ppx_syntax.wiki (hunk 4 above)
Comment (facelesspanda) [reply doc/dev/api/server/Ppx_syntax.wiki]: `let%rpc` extends to a `let%server` and a `let%client`, right? 
You already cannot nest those and they need to be top-level, so it makes sense to me.

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+
+RPC constructions are only available at top level and cannot be nested.
+ The corresponding declaration in {{{.eliomi}}} is located in shared section.
+
+==Restrictions
+
+Your RPC will implicitly compute a client part and server part. Explicit 
+declaration of each argument's monomorphic type of your RPC function will be 
+required to allow the json translation and communication between the two parts:
```
Comment (ilankri) [inline doc/dev/api/server/Ppx_syntax.wiki:33]:
```suggestion
required to allow the JSON translation and communication between the two parts:
```

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+An ocsigen-start ppx syntax extension that allows to write, in one function, both the
+client and server part of your RPC.
+
+
+==Constructions
+
+Three types of constructions are available to build an RPC:
+
+<<code language="ocaml"|
+  let%rpc toto (... = <type>) = ...
```
Comment (ilankri) [inline doc/dev/api/server/Ppx_syntax.wiki:12]:
```suggestion
  let%rpc toto (... : <type>) = ...
```

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+>>
+
+Construction to compute an RPC using the {{connected_wrapper}} API.
+
+<<code language="ocaml"|
+  let%crpc toto userid (... = <type>) = ...
```
Comment (ilankri) [inline doc/dev/api/server/Ppx_syntax.wiki:18]:
```suggestion
  let%crpc toto userid (... : <type>) = ...
```

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+
+  let%crpc_opt toto userid_o (... = <type>) = ...
```
Comment (ilankri) [inline doc/dev/api/server/Ppx_syntax.wiki:20]:
```suggestion
  let%crpc_opt toto userid_o (... : <type>) = ...
```

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+<<code language="ocaml"|
+let%rpc toto (... : <type>) = ...
+
+>>
+
+In the case of {{crpc}} and {{crpc_opt}}, as you use the {{connected_rpc}} API, 
+your first RPC function argument will refer automatically to the current session 
+user id. {{crpc_opt}} annotation refers to the case when session current id 
+needed as an optional arguement:
```
Comment (ilankri) [inline doc/dev/api/server/Ppx_syntax.wiki:43]:
```suggestion
In the case of {{crpc}} and {{crpc_opt}}, as you use the {{connected_rpc}} API, 
your first RPC function argument will refer automatically to the user ID of the
current session. {{crpc_opt}} annotation refers to the case when the user ID of the current session
is needed as an optional argument:
```

File: doc/dev/api/server/Ppx_syntax.wiki [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+
+Construction to compute an RPC using the {{connected_rpc}} API.
+
+
+RPC constructions are only available at top level and cannot be nested.
+ The corresponding declaration in {{{.eliomi}}} is located in shared section.
+
+==Restrictions
+
+Your RPC will implicitly compute a client part and server part. Explicit
```
Comment (ilankri) [inline doc/dev/api/server/Ppx_syntax.wiki:31]:
```suggestion
Your RPC will implicitly compute a client part and a server part. Explicit 
```

**Pull request merged**